app/src/main/java/com/example/fbufinalapp/EditDestinationActivity.java

```diff
@@ -69,23 +70,24 @@ protected void onCreate(Bundle savedInstanceState) {
 
             tvLocation.setFocusable(false);
 
-            // Specify the fields to return.
-            final List<Place.Field> placeFields = Arrays.asList(Place.Field.ID, Place.Field.NAME);
+            if (placeId != null){
+                // Specify the fields to return.
+                final List<Place.Field> placeFields = Arrays.asList(Place.Field.ID, Place.Field.NAME);
 
-            // Construct a request object, passing the place ID and fields array.
-            final FetchPlaceRequest request = FetchPlaceRequest.newInstance(placeId, placeFields);
+                // Construct a request object, passing the place ID and fields array.
+                final FetchPlaceRequest request = FetchPlaceRequest.newInstance(placeId, placeFields);
 
-            PlacesClient placesClient = Places.createClient(EditDestinationActivity.this);
+                PlacesClient placesClient = Places.createClient(EditDestinationActivity.this);
 
-            placesClient.fetchPlace(request).addOnSuccessListener((response) -> {
-                place = response.getPlace();
-                tvLocation.setText(place.getName());
-            }).addOnFailureListener((exception) -> {
-                if (exception instanceof ApiException) {
-                    final ApiException apiException = (ApiException) exception;
-                    Toast.makeText(EditDestinationActivity.this, "Error retrieving location", Toast.LENGTH_SHORT).show();
-                }
-            });
+                placesClient.fetchPlace(request).addOnSuccessListener((response) -> {
+                    place = response.getPlace();
+                    tvLocation.setText(place.getName());
+                }).addOnFailureListener((exception) -> {
+                    if (exception instanceof ApiException) {
```

Comment:
I see this triggers logic on an ApiException, are there any other exceptions that can get thrown and if so, what's the tradeoff of not having every exception in this case output that "Error Retrieving Location" method and then showing the exception type? 

I.e. 
```
Toast.makeText(EditDestinationActivity.this, "Error Retrieving Location", Toast.LENGTH_SHORT).show()
```

With no check for exception type

Reply:
I believe the addOnFailureListener is on the api fetch request, so its main purpose is to check when that request fails. Given this, I'm actually not sure what other types of errors it could potentially catch. Maybe if the request is set up incorrectly (which at that point is a problem with the code, and not with the user or the api request)? I can look into this!